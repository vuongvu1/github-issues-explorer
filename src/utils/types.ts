export type IssueType = Array<{
  id: string;
  title: string;
  number: number;
  createdAt: string;
  author: {
    login: string;
  };
  comments: {
    totalCount: number;
  };
}>;
