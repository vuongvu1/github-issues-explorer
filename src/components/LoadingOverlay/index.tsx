import { FC } from "react";
import { Spinner } from "src/assets/icons";
import SC from "./styles";

interface Props {
  isLoading?: boolean;
}

const LoadingOverlay: FC<Props> = ({ isLoading }) => {
  if (!isLoading) return null;

  return (
    <SC.LoadingOverlay>
      <Spinner width="48" height="48" />
    </SC.LoadingOverlay>
  );
};

export default LoadingOverlay;
