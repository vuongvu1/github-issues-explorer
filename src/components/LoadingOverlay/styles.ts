import styled, { css } from "styled-components";

const SC = {
  LoadingOverlay: styled.div(
    ({ theme }) => css`
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      background-color: ${theme.palette.boxShadow};
      display: flex;
      justify-content: center;
      align-items: center;
    `
  ),
};

export default SC;
