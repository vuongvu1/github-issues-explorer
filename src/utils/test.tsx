import { ReactNode } from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import { BrowserRouter } from "react-router-dom";
import { theme } from "src/globalStyles";
import {
  IssueState,
  IssueOrderField,
  OrderDirection,
} from "src/generated/graphql";

const mockStore = configureMockStore();
const store = mockStore({
  repoSlice: {
    url: "https://github.com/reactjs/reactjs.org",
    currentPage: 1,
    searchParams: {
      name: "reactjs.org",
      owner: "reactjs",
      status: IssueState.Open,
      orderBy: IssueOrderField.CreatedAt,
      orderDir: OrderDirection.Desc,
      before: undefined,
      after: undefined,
      first: 10,
      last: undefined,
    },
  },
});

export const customRender = (Component: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <MockedProvider mocks={[]} addTypename={false}>
        <BrowserRouter>
          <Provider store={store}>{Component}</Provider>
        </BrowserRouter>
      </MockedProvider>
    </ThemeProvider>
  );
