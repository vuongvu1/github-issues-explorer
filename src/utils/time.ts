const ONE_YEAR_IN_SECONDS = 31536000;
const ONE_MONTH_IN_SECONDS = 2592000;
const ONE_DAY_IN_SECONDS = 86400;
const ONE_HOUR_IN_SECONDS = 3600;
const ONE_MINUTE_IN_SECONDS = 60;

export const timeSince = (date: Date) => {
  const seconds = Math.floor((new Date().getTime() - date.getTime()) / 1000);

  let interval = seconds / ONE_YEAR_IN_SECONDS;

  if (interval > 1) {
    return `${Math.floor(interval)} years`;
  }

  interval = seconds / ONE_MONTH_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} months`;
  }

  interval = seconds / ONE_DAY_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} days`;
  }

  interval = seconds / ONE_HOUR_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} hours`;
  }

  interval = seconds / ONE_MINUTE_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} minutes`;
  }

  return `${Math.floor(seconds)} seconds`;
};
