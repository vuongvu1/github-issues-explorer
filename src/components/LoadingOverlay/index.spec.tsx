import { cleanup } from "@testing-library/react";
import { customRender } from "src/utils/test";
import LoadingOverlay from "./index";

describe("Text", () => {
  afterEach(cleanup);

  test("should match snapshot when loading", () => {
    const { container } = customRender(<LoadingOverlay isLoading />);

    expect(container).toMatchSnapshot();
  });

  test("should return empty when not loading", () => {
    const { container } = customRender(<LoadingOverlay isLoading={false} />);

    expect(container).toMatchSnapshot();
  });
});
