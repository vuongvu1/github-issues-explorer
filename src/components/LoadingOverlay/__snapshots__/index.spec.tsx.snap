// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot when loading 1`] = `
<div>
  <div
    class="sc-gsDJrp cmdCSZ"
  >
    <svg
      class="sc-bdvvaa fusWIN"
      height="48"
      viewBox="0 0 48 48"
      width="48"
    >
      <circle
        cx="24"
        cy="4"
        fill="#000"
        r="4"
      />
      <circle
        cx="12.19"
        cy="7.86"
        fill="#000"
        r="3.7"
      />
      <circle
        cx="5.02"
        cy="17.68"
        fill="#000"
        r="3.4"
      />
      <circle
        cx="5.02"
        cy="30.32"
        fill="#000"
        r="3.1"
      />
      <circle
        cx="12.19"
        cy="40.14"
        fill="#000"
        r="2.8"
      />
      <circle
        cx="24"
        cy="44"
        fill="#000"
        r="2.5"
      />
      <circle
        cx="35.81"
        cy="40.14"
        fill="#000"
        r="2.2"
      />
      <circle
        cx="42.98"
        cy="30.32"
        fill="#000"
        r="1.9"
      />
      <circle
        cx="42.98"
        cy="17.68"
        fill="#000"
        r="1.6"
      />
      <circle
        cx="35.81"
        cy="7.86"
        fill="#000"
        r="1.3"
      />
    </svg>
  </div>
</div>
`;

exports[`Text should return empty when not loading 1`] = `<div />`;
